spec/tapioca/cli/dsl_spec.rb

```diff
@@ -1904,6 +1935,316 @@ module GeneratedBar; end
+                  extend T::Sig
+
+                  ConstantType = type_member {{ fixed: T.class_of(Encryptable) }}
+
+                  sig { override.returns(T::Enumerable[Module]) }
+                  def self.gather_constants
+                    # Collect all the classes that include Encryptable
+                    all_classes.select { |c| c < ::Encryptable }
+                  end
+
+                  sig { override.void }
+                  def decorate
+                    # Create a RBI definition for each class that includes Encryptable
+                    root.create_path(constant) do |klass|
+                      # For each encrypted attribute we find in the class
+                      constant.__tapioca_encrypted_attributes.each do |attr_name|
+                        # Create the RBI definitions for all the missing methods
+                        klass.create_method(attr_name, return_type: "String")
+                        klass.create_method("\#{attr_name}=", parameters: [ create_param("value", type: "String") ], return_type: "void")
+                        klass.create_method("\#{attr_name}_encrypted", return_type: "String")
+                        klass.create_method("\#{attr_name}_encrypted=", parameters: [ create_param("value", type: "String") ], return_type: "void")
+                      end
+                    end
+                  end
+                end
+              end
+            end
+          RB
+
+          @project.require_mock_gem(encryptable)
+          @project.bundle_install
+
+          result = @project.tapioca("dsl CreditCard")
+
+          assert_equal(<<~OUT, result.out)
+            Loading DSL extension classes... Done
+            Loading Rails application... Done
+            Loading DSL compiler classes... Done
+            Compiling DSL RBI files...
+
+                  create  sorbet/rbi/dsl/credit_card.rbi
+
+            Done
+
+            Checking generated RBI files...  Done
+              No errors found
+
+            All operations performed in working directory.
+            Please review changes and commit them.
+          OUT
+
+          assert_project_file_equal("sorbet/rbi/dsl/credit_card.rbi", <<~RBI)
+            # typed: true
+
+            # DO NOT EDIT MANUALLY
+            # This is an autogenerated file for dynamic methods in `CreditCard`.
+            # Please instead update this file by running `bin/tapioca dsl CreditCard`.
+
+            class CreditCard
+              sig { returns(String) }
+              def number; end
+
+              sig { params(value: String).void }
+              def number=(value); end
+
+              sig { returns(String) }
+              def number_encrypted; end
+
+              sig { params(value: String).void }
+              def number_encrypted=(value); end
+            end
+          RBI
+
+          assert_empty_stderr(result)
+          assert_success_status(result)
+
+          @project.remove("sorbet/rbi/dsl")
```

Comment:
Thanks for adding tests 🙏 

Can you add one showing what happens if the extension raises an exception please?